### PR TITLE
Closes #1464 increase test coverage for derive vars aager@devel

### LINKED ISSUE #1464: General Issue: Increase test coverage for `derive_vars_aage.R`
### Background Information

We need to increase test coverage for our functions. The below screenshot has the function and gap in test coverage that we need to write additional unit tests derive_vars_aage.  Two functions in the file are deprecated and do not need tests.

image

### Definition of Done

New tests written that covers the red gap area for derive_vars_aage.R.

Use devtools::test_coverage() to reproduce the report

### PR DESCRIPTION
**Pull request by fshanlee**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation-1)?
- [ ] Update to all relevant roxygen headers and examples 
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by fshanlee** [comment]
Hi @bms63, I've added some more tests for derive_var_age_years(), as requested for issue #1464. I've written tests for what I think the function is trying to achieve, but in a couple of cases, the code would need to be updated in order to pass the tests. I was wondering what are your thoughts regarding the following: Who should update the actual function? Is it okay for me to update the function as part of this pull request, so that the function will pass the tests, or should the original developer of the function, or some other independent developer, update the function? Should my suggested changes be raised as a GitHub issue? For information, my suggested changes are:

- Line 161,    `unit <- tolower(unique(pull(dataset, !!sym(unit_var))))`, swap order of `tolower()` and `unique()`, otherwise, for example, "months" and "Months" are considered different units, so if they both appear in AGEU, then a warning will be generated even when everything else is fine.
- Add a check for age_unit, to confirm that its length is not greater than one: the message specified between lines 173 and 176 assumes that age_unit has been specified with a single value, but there is currently no check to confirm that this is true.

File: R/derive_vars_aage.R [hunk 1]
```diff
@@ -132,6 +132,13 @@ derive_var_age_years <- function(dataset, age_var, age_unit = NULL, new_var) {
   age_var <- age_variable
   unit_var <- paste0(quo_get_expr(age_var), "U")
 
+  if (!is.null(age_unit)) {
+    if (length(age_unit) > 1) {
+      err_msg <- "Only one unit can be specified via the `age_unit` parameter"
+      abort(err_msg)
+    }
+  }
+
```
Comment (bms63) [inline R/derive_vars_aage.R:141]: Can we use any of our `assert` functions here?  https://pharmaverse.github.io/admiraldev/devel/reference/index.html

File: tests/testthat/test-derive_vars_aage.R [hunk 2]
```diff
@@ -51,3 +51,89 @@ test_that("derive_var_age_years - Error is issued if age_unit is missing", {
     derive_var_age_years(input, AGE, new_var = AAGE)
   )
 })
+
+test_that(paste0("derive_var_age_years - Warning is issued if age_unit is not null, but the ",
```
Comment (bms63) [inline tests/testthat/test-derive_vars_aage.R:55]: I think you can just put these on new lines and remove the paste0.  Also you can use  `#nolint` to the end of the line and the lintr won't checkt hat line.  Don't use it here, but just FYI

Comment (bms63) [reply tests/testthat/test-derive_vars_aage.R:55]: Can you number these tests following the guidance here.  There is an addin in `admiraldev` that can do this for you. 

https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#set-up-the-test-script

https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#addin-admiraldevformat_test_that_file

**Review by bms63 — CHANGES_REQUESTED** [review]
So happy your back in admiral!! This is a great find Shan and I appreciate your attention to detail here!!!!

File: tests/testthat/test-derive_vars_aage.R (hunk 2 above)
Comment (fshanlee) [reply tests/testthat/test-derive_vars_aage.R:55]: > I think you can just put these on new lines and remove the paste0. Also you can use `#nolint` to the end of the line and the lintr won't checkt hat line. Don't use it here, but just FYI

When I remove `paste0()`, then I can only specify a single text string: I can't specify `"abc" "def"`, nor can I specify `"abc", "def"`; I can only specify `"abc _<new line>_ def"`. If I then force the testthat script to fail, then the message will be split wherever I have a new line in the testthat script:

`Failure (test-derive_vars_aage.R:68:3): derive_var_age_years - Warning is issued if age_unit is not null, but the 'unit' variable corresponding to age_var stores more than one unique value.`
 
, becomes

```
Failure (test-derive_vars_aage.R:67:3): derive_var_age_years - Warning is issued if age_unit is not null, but the 'unit'
           variable corresponding to age_var stores more than one unique value.

```
So I think the messages are neater when I used paste0(). If there is another way to split the text in the testthat script without resulting in unnecessary newlines in the actual messages, then please let me know.
BTW, the above example doesn't look so bad when I copy and paste into GitHub, but in the R Studio console, the text `unit` appears near the start of a new line, so it looks odd when `variable corresponding to` starts on the next line.

File: R/derive_vars_aage.R (hunk 1 above)
Comment (fshanlee) [reply R/derive_vars_aage.R]: Thanks for your suggestion. I've moved the call to **assert_character_scalar()**, previously at line 146, to nearer the start of the function, and I used that to replace the code that I wrote to check **age_unit**. I also amended the call to **assert_character_scalar()**, assigning values to the parameters **case_sensitive** and **optional**.
There is possibly a bug with **assert_character_scalar()**, which I've raised as issue number 156 in {admiraldev}, though I'm no longer able to reproduce the issue.

**Comment by bms63** [comment]
Hey @fshanlee 

I did a test on your branch.  I removed the `paste()` function and just put the message on the new line and it works.  There are other issues going with the tests, but I think this works as expected.

image

**Comment by fshanlee** [comment]
Hi @bms63 , do the workflows for {admiral} use the most recent _devel_ branch for {admiraldev}? The checks for this pull request are failing because the function `assert_date_vector() `could not be found, but I can confirm that this function is defined on the _devel_ branch of {admiraldev}, even though it's not defined on the _Main_ branch of {admiraldev}. Would it be easy to change the workflows for {admiral}, to install {admiraldev} from its _devel_ branch, before running R CMD check for {admiral}? I'm wondering whether this has caused a problem for anyone else submitting a pull request.

**Comment by bms63** [comment]
oh you need to have `@devel` at the end of your branch so the actions know to look at the devel branches for `admiral,test` and `admiraldev`.  Without it they just point at `main`

File: tests/testthat/test-derive_vars_aage.R (hunk 2 above)
Comment (fshanlee) [reply tests/testthat/test-derive_vars_aage.R]: As discussed during a team meeting, I've removed the calls to paste0() when I call test_that().

**Comment by fshanlee** [comment]
Hi @bms63 , I've created a branch called "1464... @devel", which is based on the branch "1464... ". When I tried to change the branch name in this pull request, I think I just changed the name of this pull request, so I'm going to close this pull request now, and open a new pull request for the new "1464... @devel" branch.

**Pull request closed without merging**